File: Gifs_Watcher/app/src/main/java/com/example/gifs_watcher/services/TenorService.java
package com.example.gifs_watcher.services;


import com.example.gifs_watcher.models.TenorData;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface TenorService {

    @GET("search/?dataset=sites-2g-3g-4g-france-metropolitaine-mon-reseau-mobile&q=&sort=code_op")
    Call<TenorData> getPaginateDataSorted(@Query("lang") String lang, @Query("rows") String rows, @Query("start") String start);

    @GET("search/?dataset=sites-2g-3g-4g-france-metropolitaine-mon-reseau-mobile&q=")
    Call<TenorData> getPaginateData(@Query("lang") String lang, @Query("rows") String rows, @Query("start") String start);
}

File: Gifs_Watcher/app/src/main/java/com/example/gifs_watcher/services/TenorService.kt
package com.example.gifs_watcher.services

import com.example.gifs_watcher.models.TenorData
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

interface TenorService {
    @GET("/v1/random?key=LIVDSRZULELA&limit=10&contentfilter=high")
    fun getRandomsGifs(
        @Query("key") key: String?,
        @Query("limit") limit: String?,
        @Query("contentfilter") contentfilter: String?
    ): Call<TenorData?>?
}